apps/web/src/stores/commitContext.tsx

```diff
@@ -0,0 +1,81 @@
+import {
+  createContext,
+  ReactNode,
+  useContext,
+  useEffect,
+  useState,
+} from 'react'
+
+import { checkCommitEditableAction } from '$/actions/commits'
+
+interface CommitContextType {
+  currentCommit?: string
+  isCommitEditable: boolean
+  setCurrentCommit: ({
+    commitUuid,
+    isEditable,
+  }: {
+    commitUuid?: string
+    isEditable: boolean
+  }) => void
+}
+
+const CommitContext = createContext<CommitContextType>({
+  currentCommit: undefined,
+  isCommitEditable: false,
+  setCurrentCommit: () => {},
+})
+
+export const CommitProvider = ({ children }: { children: ReactNode }) => {
+  const [currentCommit, setCurrentCommit] = useState<string | undefined>()
+  const [isCommitEditable, setIsCommitEditable] = useState<boolean>(false)
+
+  useEffect(() => {
+    const storedCommit = localStorage.getItem('commit')
+    if (!storedCommit) {
+      setCurrentCommit(undefined)
+      setIsCommitEditable(false)
+      return
+    }
+
+    setCurrentCommit(storedCommit)
+    checkCommitEditableAction({ commitUuid: storedCommit }).then((res) => {
+      setIsCommitEditable(res[0]!)
+    })
+  }, [])
+
+  const updateCurrentCommit = ({
+    commitUuid,
+    isEditable,
+  }: {
+    commitUuid?: string
+    isEditable: boolean
+  }) => {
+    setCurrentCommit(commitUuid)
+    setIsCommitEditable(isEditable)
```

Comment:
this is derived state so it should be a memoized computation, don't duplicate the state.